### DISCUSSION TITLE
1차 기획안 작성 완료

### DISCUSSION THREAD

**Discussion by Joseph-Cha** [Joseph-Cha/JustFight_Client#4] [General]
일단 간략하게 개발기간, 시나리오, 구현범위(클라이언트 한정)까지 작성했습니다~
피드백 주시면 주말동안 추가하겠습니다😊

[기획안](https://www.notion.so/2D-096c5bbe9ab04b899bebc1ec25401357)

**Comment by jongfeel** [comment]
대략적인 기획이 나온 것 같습니다. 
실제 사용자 시나리오를 만들어 보면서 구체적으로 기획을 해 보고
기존 기획서의 부족한 부분을 보완하는 식으로 해보면 좋을 것 같습니다.

---

추가 의견
피드백을 달라고 요청하기 보다는 md 파일로 만들고 pull request를 통해 자연스럽게 리뷰가 되는 방식으로 하면 어떨까요?
물론 노션도 좋은 툴이고 히스토리 추적도 되지만 md 파일로 관리하는게 더 나을 것 같아요.

**Comment by jongfeel** [comment]
1. 인터페이스 위주로 설계서 작성

실제 코드로 만드는 인터페이스 (ex. IDamage)를 고려하지 않고
객체들끼리 어떤 데이터를 주고 받고 함수 호출을 해서 상태를 변경하는지에 대한 부분을
완벽하게 하지 말고 대략적으로 만들어 두고 시작할 수 있게 해도 충분합니다.

2. 리뷰 작전

같은 파일에 대한 지속적인 수정 => 하나의 PR에 여러번 commit을 해서 파일이 어떻게 변화가 일어나는지 리뷰
다른 파일들에 대한 별개의 수정 => 여러개의 branch에 대한 PR을 만들어서 각각 리뷰 진행

**Reply by jongfeel** [reply]
추가 아이디어

캐릭터가 죽기 까지의 시간을 재서 랭킹 스코어 보드가 있으면 게임 할 때 마다 도전의식 같은게 생길 수 있음
